crypto.c

```diff
@@ -176,6 +178,28 @@ smtp_init_crypto(int fd, int feature, struct smtp_features* features)
 		return (1);
 	}
 
+#ifdef SSL_CTRL_SET_TLSEXT_HOSTNAME
```

Comment:
that macro was introduced 18 years ago.  I think we should expect it to be present.